File: C/2846.c
#include <stdio.h>

int fibonot(int k);

int main(){

    int k;

    scanf("%i", &k);

    printf("%i\n", fibonot(k));

    return 0;
}

int fibonot(int k){
    int anterior = 1, atual = 2, proximo = 3;

    while(k > 0){
        anterior = atual;
        atual = proximo;
        proximo = anterior + atual;

        k -= (anterior - 1);
    }

    k += (anterior - 1);

    return atual + k;
}

File: Python/2846.py
def fibonot(k):
    anterior = 1
    atual = 2
    proximo = 3
    while k > 0:
        # Anda pela sequencia de fibonacci até que k seja negativo
        anterior = atual
        atual = proximo
        proximo = anterior + atual

        #Subtrai o anterior - 1 de k
        k -= (anterior - 1)

    #Soma novamente o anterior e subtrai 1
    k += (anterior - 1)

    return atual + k

k = int(input())
print(fibonot(k))